homeassistant/components/ssdp/__init__.py

```diff
@@ -85,6 +86,29 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+_SsdpServiceInfoBase = TypedDict(
+    "_SsdpServiceInfoBase",
+    {
+        "x-homeassistant-matching-domains": set,
```

Comment:
This should be `set[str]` but it generates `error: "set" is not subscriptable  [misc]`

I can adjust to this, but I am not sure it makes it better.
```suggestion
        "x-homeassistant-matching-domains": set[str],  # type: ignore[misc]
```